### PR TITLE
Adding RSI example 

### PR DESCRIPTION
**Pull request by harayuanwang**

### CODE REVIEW

**Review by jeandelavarene — COMMENTED** [review]
Add a copyright to all Java files and Javadoc as well.

**Review by Kuassim — APPROVED** [review]

**Comment by harayuanwang** [comment]
Working on the copyright and JavaDoc, please do not approve this PR before they're done.

File: java/rsi-example/src/main/java/rsi/example/amqp/Listener.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+public class Listener {
+  private static final String ACTIVEMQ_USER = "admin";
+  private static final String ACTIVEMQ_PASSWORD = "password";
+  private static final String ACTIVEMQ_HOST = "localhost";
+  private static final int ACTIVEMQ_PORT = 5672;
+  private static final String TOPIC_NAME = "event";
+
+  private static final String DB_URL = "jdbc:oracle:thin:@" +
+      "(description= (retry_count=20)(retry_delay=3)(address=(protocol=tcps)(port=1521)(host=adb.us-phoenix-1.oraclecloud.com))(connect_data=(service_name=gebqqvpozhjbqbs_azuretestdb_high.adb.oraclecloud.com))(security=(ssl_server_dn_match=yes)))";
```
Comment (gvenzl) [inline java/rsi-example/src/main/java/rsi/example/amqp/Listener.java:22]: This one should be templated to allow users to fill in their own connection details.
Certainly, we should not have a pre-existing service name in our examples.

File: java/rsi-example/src/main/java/rsi/example/mqtt/Listener.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+public class Listener {
+  private static final String ACTIVEMQ_HOST = "localhost";
+  private static final int ACTIVEMQ_PORT = 1883;
+  private static final String ACTIVEMQ_USER = "admin";
+  private static final String ACTIVEMQ_PASSWORD = "password";
+  private static final String DESTINATION = "/topic/event";
+  private static final String TOPIC_NAME = "event";
+
+  private static final String DB_URL = "jdbc:oracle:thin:@" +
+      "(description= (retry_count=20)(retry_delay=3)(address=(protocol=tcps)(port=1521)(host=adb.us-phoenix-1.oraclecloud.com))(connect_data=(service_name=gebqqvpozhjbqbs_azuretestdb_high.adb.oraclecloud.com))(security=(ssl_server_dn_match=yes)))";
```
Comment (gvenzl) [inline java/rsi-example/src/main/java/rsi/example/mqtt/Listener.java:24]: Should be templated

File: java/rsi-example/src/main/java/rsi/example/stomp/Listener.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+public class Listener {
+  private static final String ACTIVEMQ_HOST = "localhost";
+  private static final int ACTIVEMQ_PORT = 61613;
+  private static final String DESTINATION = "/topic/event";
+
+  private static final String DB_URL = "jdbc:oracle:thin:@" +
+      "(description= (retry_count=20)(retry_delay=3)(address=(protocol=tcps)(port=1521)(host=adb.us-phoenix-1.oraclecloud.com))(connect_data=(service_name=gebqqvpozhjbqbs_azuretestdb_high.adb.oraclecloud.com))(security=(ssl_server_dn_match=yes)))";
```
Comment (gvenzl) [inline java/rsi-example/src/main/java/rsi/example/stomp/Listener.java:23]: Should be templated

**Review by gvenzl — CHANGES_REQUESTED** [review]
Please template the connection strings, they currently have real service names in them.

**Comment by harayuanwang** [comment]
All the comments have been addressed

**Comment by gvenzl** [comment]
Thanks @harayuanwang. Good to go from my side.

**Pull request merged**